src/tests/unit/tests/DetailsView/components/adhoc-tab-stops-test-view.test.tsx

```diff
@@ -2,30 +2,118 @@
 
 describe('AdhocTabStopsTestView', () => {
-    const visualizationScanResultDataStub = {} as VisualizationScanResultData;
-
-    const props = {
-        configuration: {
-            displayableData: {
-                title: 'test title',
-            } as DisplayableVisualizationTypeData,
-        },
-        visualizationScanResultData: visualizationScanResultDataStub,
-        deps: {} as AdhocTabStopsTestViewDeps,
-    } as AdhocTabStopsTestViewProps;
-
-    it('renders with content', () => {
-        const rendered = shallow(<AdhocTabStopsTestView {...props} />);
-        expect(rendered.getElement()).toMatchSnapshot();
+    let props: AdhocTabStopsTestViewProps;
+    let getStoreDataMock: IMock<(data: TestsEnabledState) => ScanData>;
+    let clickHandlerFactoryMock: IMock<DetailsViewToggleClickHandlerFactory>;
+    let displayableDataStub: DisplayableVisualizationTypeData;
+    let scanDataStub: ScanData;
+    let clickHandlerStub: (event: any) => void;
+    let visualizationStoreDataStub: VisualizationStoreData;
+    let selectedTest: VisualizationType;
+    let featureFlagStoreDataStub: FeatureFlagStoreData;
+    let visualizationScanResultData: VisualizationScanResultData;
+
+    beforeEach(() => {
+        getStoreDataMock = Mock.ofInstance(() => null, MockBehavior.Strict);
+        clickHandlerFactoryMock = Mock.ofType(
+            DetailsViewToggleClickHandlerFactory,
+            MockBehavior.Strict,
+        );
+        displayableDataStub = {
+            title: 'test title',
+            toggleLabel: 'test toggle label',
+        } as DisplayableVisualizationTypeData;
+        scanDataStub = {
+            enabled: true,
+        };
+        visualizationStoreDataStub = {
+            tests: {},
+        } as VisualizationStoreData;
+        clickHandlerStub = () => {};
+        selectedTest = -1;
+        featureFlagStoreDataStub = {};
+        visualizationScanResultData = { tabStops: {} } as VisualizationScanResultData;
+
+        props = {
+            configuration: {
+                getStoreData: getStoreDataMock.object,
+                displayableData: displayableDataStub,
+            },
+            clickHandlerFactory: clickHandlerFactoryMock.object,
+            visualizationStoreData: visualizationStoreDataStub,
+            selectedTest,
+            featureFlagStoreData: featureFlagStoreDataStub,
+            visualizationScanResultData,
+            deps: Mock.ofType<AdhocTabStopsTestViewDeps>().object,
+        } as AdhocTabStopsTestViewProps;
+
+        getStoreDataMock
+            .setup(gsdm => gsdm(visualizationStoreDataStub.tests))
+            .returns(() => scanDataStub)
+            .verifiable();
+
+        clickHandlerFactoryMock
+            .setup(chfm => chfm.createClickHandler(selectedTest, !scanDataStub.enabled))
+            .returns(() => clickHandlerStub)
+            .verifiable();
+    });
+
+    it('should return target page changed view as tab is changed', () => {
```

Comment:
This uses the tabStoreData; we should either remove this test case here or not remove tabStoreData and get this test case to be correct since currently it does not return the target page changed view in the snap shot.